### PR TITLE
Add Neo4j resource manager for integration tests

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

**Comment by bvolpato** [comment]
Thanks! I will go through it soon.

A bit annoying, but it seems that there are some issues with spotless that didn't allow the rest of the tests run.

File: it/neo4j/src/test/java/com/google/cloud/teleport/it/neo4j/Neo4jResourceManagerIT.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+/** Integration tests for {@link Neo4jResourceManager}. */
+@Category(TestContainersIntegrationTest.class)
+@RunWith(JUnit4.class)
+public class Neo4jResourceManagerIT {
+
+  private Neo4jResourceManager neo4jResourceManager;
+
+  @Before
+  public void setUp() {
+    neo4jResourceManager = Neo4jResourceManager.builder("dummy").setDatabaseName("neo4j").build();
```
Comment (bvolpato) [inline it/neo4j/src/test/java/com/google/cloud/teleport/it/neo4j/Neo4jResourceManagerIT.java:40]: Can you change `dummy` to something else? Some background https://developers.google.com/style/word-list#letter-d

Comment (fbiville) [reply it/neo4j/src/test/java/com/google/cloud/teleport/it/neo4j/Neo4jResourceManagerIT.java]: Fixed in 856d953740afeb1dd9050a0e39a052158cdf4dff

**Comment by fbiville** [comment]
> A bit annoying, but it seems that there are some issues with spotless that didn't allow the rest of the tests run.

Fixed in 8f4427615add76c9640e3c46dca1585055a1ad17

**Pull request merged**